azkaban-web-server/src/main/java/cloudflow/servlets/ProjectServlet.java

```diff
@@ -0,0 +1,100 @@
+package cloudflow.servlets;
+
+import azkaban.server.HttpRequestUtils;
+import azkaban.server.session.Session;
+import azkaban.user.User;
+import azkaban.webapp.AzkabanWebServer;
+import azkaban.webapp.servlet.LoginAbstractAzkabanServlet;
+import cloudflow.services.ProjectService;
+import cloudflow.models.Project;
+import com.linkedin.jersey.api.uri.UriTemplate;
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import javax.servlet.ServletConfig;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import org.apache.http.HttpStatus;
+import org.codehaus.jackson.map.ObjectMapper;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class ProjectServlet extends LoginAbstractAzkabanServlet {
+
+  private static final String GET_ALL_PROJECT_URI = "/projects";
+  private static final String PROJECT_ID_KEY = "projectId";
+  private static final UriTemplate GET_PROJECT_URI_TEMPLATE = new UriTemplate(
+      String.format("/projects/{%s}", PROJECT_ID_KEY));
+
+  private ProjectService projectService;
+  private ObjectMapper objectMapper;
+
+  private static final Logger log = LoggerFactory.getLogger(ProjectServlet.class);
+
+  @Override
+  public void init(final ServletConfig config) throws ServletException {
+    super.init(config);
+    final AzkabanWebServer server = (AzkabanWebServer) getApplication();
+    this.projectService = server.projectService();
+    this.objectMapper = server.objectMapper();
+  }
+
+  @Override
+  protected void handleGet(HttpServletRequest req, HttpServletResponse resp, Session session)
+      throws IOException, ServletException {
+    Map<String, String> templateVariableToValue = new HashMap<>();
+    if (GET_ALL_PROJECT_URI.equals(req.getRequestURI())) {
+      /* Get all records */
+      String response = objectMapper.writeValueAsString(projectService.getAllProjects(session.getUser()));
+      this.writeResponse(resp, response);
```

Comment:
Please use the utility method `sendResponse` instead so that HTTP status code and Content Type are set in the response.

Reply:
Fixed.